infra/cdk/lib/stacks/env/main/mainCertificates.ts

```diff
@@ -0,0 +1,56 @@
+import {CfnOutput, Construct} from "@aws-cdk/core";
+import {DnsValidatedCertificate} from '@aws-cdk/aws-certificatemanager';
+import {PublicHostedZone} from "@aws-cdk/aws-route53";
+
+import {EnvConstructProps} from "../../../types";
+import {EnvironmentSettings} from "../../../settings";
+
+
+export interface MainCertificateProps extends EnvConstructProps {
+}
+
+export class MainCertificates extends Construct {
+    certificate: DnsValidatedCertificate;
+    cloudFrontCertificate: DnsValidatedCertificate;
+
+    static geCertificateArnOutputExportName(envSettings: EnvironmentSettings) {
+        return `${envSettings.projectEnvName}-certificateArn`
+    }
+
+    static geCloudFrontCertificateArnOutputExportName(envSettings: EnvironmentSettings) {
+        return `${envSettings.projectEnvName}-cfCertificateArn`
+    }
+
+    constructor(scope: Construct, id: string, props: MainCertificateProps) {
+        super(scope, id);
+
+        const hostedZone = PublicHostedZone.fromHostedZoneAttributes(this, "DomainZone", {
+            hostedZoneId: props.envSettings.hostedZone.id,
+            zoneName: props.envSettings.hostedZone.name,
+        });
+
+        const domainName = `${props.envSettings.envStage}.${props.envSettings.hostedZone.name}`
+        this.certificate = new DnsValidatedCertificate(this, "AppsCertificate", {
```

Comment:
when creating this stack cdk will hang until we add dns to the domain?

Reply:
No, this construct will create the validation records in the referenced hosted zone automatically, which is awesome.

Reply:
nice!